### PR TITLE
[DEV-3709] handing application lifecycle

### PR DESCRIPTION
**Pull request by bushwhackr**
## Description

+ Manage websocket closure properly
+ Convert fastapi startup/shutdown events to the new fastapi syntax
+ Linting task fixes

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/app.py [hunk 1]
```diff
@@ -126,6 +128,75 @@ async def _wrapper(
...
+                logger.debug("Websocket disconnected", extra={"exception": e})
+                if websocket.client_state == WebSocketState.CONNECTED:
+                    await websocket.close(code=status.WS_1000_NORMAL_CLOSURE)
+            except Exception as e:
+                logger.error("Websocket connection error", extra={"exception": e})
+                # Close client connection if still active
+                if websocket.client_state == WebSocketState.CONNECTED:
+                    await websocket.close(code=status.WS_1011_INTERNAL_ERROR)
+            finally:
+                ws_connections.remove(websocket)
```
Comment (kchua78) [inline featurebyte/app.py:176]: WIll doing the close in finally not get triggered on shutdown?

Comment (kchua78) [reply featurebyte/app.py:176]: We need to add `global ws_connections` here to use the variable.

Comment (bushwhackr) [reply featurebyte/app.py:176]: Will put the global in. 

Im not sure the order of operations for the server shutdown, whether the lifecycle stop fires first; closing the connection, or the finally block gets triggered by an exception.
Im just putting both sides as a safeguard.

Comment (bushwhackr) [reply featurebyte/app.py:176]: @kchua78  I've looked at the global part. 
I'm not mutating the ws_connections pointer, so it should work without adding the `global`.

**Review by kchua78 — APPROVED** [review]

**Pull request merged**